### PR TITLE
Add Whatsapp sharing and fix WebView load bug

### PR DESCRIPTION
**Pull request by dgtv**
## Related Stories

[](https://talkable.atlassian.net/browse/PR-13466)
[](https://talkable.atlassian.net/browse/PR-13440)

### CODE REVIEW

File: sdk/src/main/java/com/talkable/sdk/TalkableOfferFragment.java [hunk 1]
```diff
@@ -138,6 +139,10 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == REQUEST_CODE_SEND_NATIVE_MAIL && resultCode == Activity.RESULT_OK) {
             shareSucceeded(SharingChannel.NATIVE_MAIL.toString());
         }
+
+        if (requestCode == REQUEST_CODE_SEND_WHATSAPP) {
```
Comment (zhuravel) [inline sdk/src/main/java/com/talkable/sdk/TalkableOfferFragment.java:143]: Should there be a resultCode check?

File: sdk/src/main/java/com/talkable/sdk/WebAppInterface.java [hunk 2]
```diff
@@ -71,6 +71,10 @@ public void publish(String eventName, String data) {
                     talkableOfferFragment.shareOfferViaTwitter(
                             JsonUtils.getJsonString(json, "message"));
                     break;
+                case "share_offer_via_whatsapp":
+                    talkableOfferFragment.shareOfferViaWhatsapp(
+                            JsonUtils.getJsonString(json, "message"));
```
Comment (zhuravel) [inline sdk/src/main/java/com/talkable/sdk/WebAppInterface.java:76]: Why claim_url is not passed for Twitter and WhatsApp, but passed for other channels?

Comment (dgtv) [reply sdk/src/main/java/com/talkable/sdk/WebAppInterface.java:76]: It is a question not to the SDK but to `talkable.js`. It passes only `message` for that sharing methods.

Comment (dgtv) [reply sdk/src/main/java/com/talkable/sdk/WebAppInterface.java:76]: And actually message contains claim URL. Here is a sample message from default theme: `Get $5 off your first order at dgt-store https://test-prod2.automatic-tests-talkable.com/x/6lLb5Z`

File: sdk/src/main/java/com/talkable/sdk/TalkableOfferFragment.java (hunk 1 above)
Comment (dgtv) [reply sdk/src/main/java/com/talkable/sdk/TalkableOfferFragment.java]: Updated

File: sdk/src/main/java/com/talkable/sdk/TalkableOfferFragment.java [hunk 3]
```diff
@@ -323,12 +328,28 @@ public void shareOfferViaFacebookMessage(String claimUrl) {
         FacebookUtils.shareViaMessenger(this, claimUrl);
     }
 
+    public void shareOfferViaWhatsapp(String message) {
```
Comment (zhuravel) [inline sdk/src/main/java/com/talkable/sdk/TalkableOfferFragment.java:331]: Naming should be consistent between this and `isWhatsAppAvailable`.

```suggestion
    public void shareOfferViaWhatsApp(String message) {
```

**Pull request merged**